File: Golang/test/contain_duplicate.go
package test

func ContainDuplicate(nums []int) bool { // [1, 2, 3, 4]
	m := make(map[int]int)

	for _, num := range nums {
		if _, found := m[num]; found {
			return true
		}

		m[num] = num
	}

	return false
}

File: Javascript/day_3_hashmap_and_two_pointer/containDuplicate.js
function containDuplicate(nums) {
  const numbersMap = new Set();

  for (const num of nums) {
    if (numbersMap.has(num)) {
      return true;
    }

    numbersMap.add(num);
  }

  return false;
}

console.time()
console.log(containDuplicate([1, 2, 3, 4]));
console.log(containDuplicate([1, 2, 3, 4, 1]));
console.timeEnd()
